src/services/user/services/AdminUsers.js

```diff
@@ -87,7 +87,24 @@ class AdminUsers {
 
 			// fetch data that are scoped to schoolId
 			const searchedRole = roles.find((role) => role.name === this.role);
-			return getAllUsers(schoolId, currentYear, searchedRole._id, query);
+			const users = await getAllUsers(schoolId, currentYear, searchedRole._id, query);
+
+			// filter students' latest class and return students filtered by class query
+			if (params.headers.referer.includes('students')) {
+				users.data.forEach((student) => {
+					student.classes = student.classes.sort(
+						(a, b) => b.match(/\d+/) - a.match(/\d+/),
```

Comment:
I'm not sure what `student.classes` refers to here, but I doubt this will work for class names without numbers ("Froschklasse")